x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -803,7 +803,7 @@ public double getSamplingRate() {
             return samplingRate;
         }
 
-        public void setRequestedDuration(double requestedDuration) {
+        public void setRequestedDuration(Double requestedDuration) {
```

Comment:
Curious, how does this change fix a null pointer exception? I'd assume that `this.requestedDuration = requestedDuration;` would never assign `null` to `this.requestedDuration`, regardless of what `double` value is passed in.

Reply:
`Double` is an object while `double` is a basic type.
When assigning a `Double` to a `double`, under the hood `Double.doubleValue()` is called. So if the `Double` value is `null`, you get a null pointer exception.

Previously, when you called `setRequestedDuration(null)` the exception happened, at the call site, not inside the function.

The `getRequestedDuration()` checks for the `null` value and has a fallback installed.
Since the API is only used by Kibana is not publically announced, the `requested_duration` request option is always sent ... well, except yesterday where there was a bug in Kibana which caused an exception :)